### ISSUE TITLE
Expose ldap filter for users as configuration item

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#2438]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/2434
- Created at 2014-09-09 23:02:26 by [bnordgren](https://pagure.io/user/bnordgren)
- Closed at 2020-03-24 14:13:09 as wontfix
- Assigned to nobody
---

Certain deployment scenarios may call for granting end users (system administrators) control over the ldap query used to locate users in an ldap backend. Scenarios mentioned on the discussion list include:

1. Heavy use of fine-grained ACLs
1. Use in an environment where a local ldap server proxies (and may locally override) user attributes from a remote identity store.

The current base filter is:
 

    (&(objectclass=posixAccount)(uid=*)(&(uidNumber=*)(!(uidNumber=0))))

When searching for a uid given a uidNumber, the query becomes: 

    (&(uidNumber=$key)(objectclass=posixAccount)(uid=*)(&(uidNumber=*)(!(uidNumber=0))))

Where the first term makes the last two terms moot. In addition, with certain configurations of an ldap proxy, the (uid=*) term can cause the proxy to request all user records from the proxied server. 

Suggest exposing this filter to the local system administrator. Code will need to check the returned entry for the presence of required attributes and apply any necessary constraints to the attribute values.

### Email threads

This has come up in the discussion list at least twice: 

1. https://lists.fedorahosted.org/pipermail/sssd-users/2014-May/001630.html
1. https://lists.fedorahosted.org/pipermail/sssd-users/2014-September/002181.html

### Comments

---

**Comment from [bnordgren](https://pagure.io/user/bnordgren) at [2014-09-16 10:51:54](https://pagure.io/SSSD/sssd/issue/2434#233874)**

This might make sense in the context of #2038

milestone: NEEDS_TRIAGE => SSSD 1.13 beta

---

**Comment from [bnordgren](https://pagure.io/user/bnordgren) at [2014-09-16 10:54:41](https://pagure.io/SSSD/sssd/issue/2434#233875)**

Fields changed

blocking:  => 2038

---

**Comment from [bnordgren](https://pagure.io/user/bnordgren) at [2014-09-26 18:50:00](https://pagure.io/SSSD/sssd/issue/2434#233876)**

Fields changed

mark:  => 0

---

**Comment from [bnordgren](https://pagure.io/user/bnordgren) at [2014-11-05 14:13:04](https://pagure.io/SSSD/sssd/issue/2434#233877)**

Fields changed

rhbz:  => todo

---

**Comment from [bnordgren](https://pagure.io/user/bnordgren) at [2015-02-10 16:10:21](https://pagure.io/SSSD/sssd/issue/2434#233878)**

Nice to have.

milestone: SSSD 1.13 beta => SSSD 1.13 backlog

---

**Comment from [bnordgren](https://pagure.io/user/bnordgren) at [2015-02-12 20:13:41](https://pagure.io/SSSD/sssd/issue/2434#233879)**

Mass-moving tickets not planned for the 1.13 release to 1.14

milestone: SSSD 1.13 backlog => SSSD 1.14 beta

---

**Comment from [bnordgren](https://pagure.io/user/bnordgren) at [2015-07-28 20:08:32](https://pagure.io/SSSD/sssd/issue/2434#233880)**

Fields changed

priority: major => minor
sensitive:  => 0

---

**Comment from [bnordgren](https://pagure.io/user/bnordgren) at [2016-02-16 14:42:48](https://pagure.io/SSSD/sssd/issue/2434#233881)**

This makes sense, but the 1.14 milestone is too big. I'm moving the ticket to 1.15, in the meantime, patches are welcome.

milestone: SSSD 1.14 beta => SSSD 1.15 beta

---

**Comment from [bnordgren](https://pagure.io/user/bnordgren) at [2017-02-24 14:48:31](https://pagure.io/SSSD/sssd/issue/2434#233882)**

**Metadata Update from @bnordgren**:
- Issue marked as blocked by: #2038
- Issue set to the milestone: SSSD Future releases (no date set yet)

---

**Comment from [bnordgren](https://pagure.io/user/bnordgren) at [2020-03-13 12:25:35](https://pagure.io/SSSD/sssd/issue/2434#633121)**

**Metadata Update from @thalman**:
- Custom field blocking reset (from 2038)
- Custom field design_review reset (from 0)
- Custom field mark reset (from 0)
- Custom field patch reset (from 0)
- Custom field review reset (from 0)
- Custom field sensitive reset (from 0)
- Custom field testsupdated reset (from 0)
- Issue close_status updated to: None
- Issue tagged with: Canditate to close

---

**Comment from [bnordgren](https://pagure.io/user/bnordgren) at [2020-03-24 14:13:09](https://pagure.io/SSSD/sssd/issue/2434#636720)**

Thank you for taking time to submit this request for SSSD. Unfortunately this issue was not given priority and the team lacks the capacity to work on it at this time.

Given that we are unable to fulfill this request I am closing the issue as wontfix.

If the issue still persist on recent SSSD you can request re-consideration of this decision by reopening this issue. Please provide additional technical details about its importance to you.

Thank you for understanding.

---

**Comment from [bnordgren](https://pagure.io/user/bnordgren) at [2020-03-24 14:13:10](https://pagure.io/SSSD/sssd/issue/2434#636721)**

**Metadata Update from @pbrezina**:
- Issue close_status updated to: wontfix
- Issue status updated to: Closed (was: Open)
